daemon/src/engine/client/cl_main.cpp

```diff
@@ -1429,7 +1428,67 @@ void CL_Connect_f()
 	Cvar_Set( "cl_currentServerIP", serverString );
 }
 
-static const int MAX_RCON_MESSAGE = 1024;
+static void CL_RconSend(const Rcon::Message &message)
+{
+	std::string invalid_reason;
+	if ( message.valid(&invalid_reason) )
+	{
+		message.send();
+	}
+	else
+	{
+		Log::Notice("Invalid rcon message: %s\n", invalid_reason.c_str());
+	}
+}
+
+struct RconChallengeQueue
```

Comment:
Also why do we need locking here? At this point of the read I assume this will always be called from the main thread